src/components/FileUpload.js

```diff
@@ -87,44 +86,82 @@ const FileUpload = React.createClass({
 
         this.setState({
           dragging: false,
-          imageSource,
-          invalidMessage: null
+          imageSource
         });
-        this.props.onFileAdd(file);
       };
     }
   },
 
-  _removeImage (e) {
+  _removeFile (e) {
     e.stopPropagation();
     e.preventDefault();
 
     this.setState({
-      imageSource: null
+      imageSource: null,
+      invalidMessage: null
     });
     this.props.onFileRemove();
   },
 
+  _renderInvalidMessage () {
+    if (this.state.invalidMessage) {
+      return (
+        <div style={styles.invalidMessage}>
+          <Icon style={styles.invalidIcon} type='attention' />
+          {this.state.invalidMessage}
+        </div>
+      );
+    } else {
+      return null;
+    }
+  },
+
+  _validateFile (file) {
+    const isTooBig = this.props.maxFileSize < file.size / 1000;
+    const isInvalidType = this.props.allowedFileTypes && this.props.allowedFileTypes.indexOf(file.type) < 0;
+
+    if (isTooBig || isInvalidType) {
```

Comment:
I'm finding the nested ternary inside of the if a little complex. What if you use a let statement to set the message in each case and then setState after with whatever the value is?